### ISSUE TITLE
`anndata.concat()` merging of `uns` doesn't work for `None` values

### ISSUE THREAD

**Issue by alam-shahul** [scverse/anndata#711]
I'm trying to use `anndata.concat()` to merge two AnnData objects with the following structure:

```
object_1 = {
...
uns: {
    "property": {
        "0": None
    }
}
...
}

object_2 = {
...
uns: {
    "property": {
        "1": None
    }
}
...
}
```

When I merge them using `uns_merge="unique"`, the `"property"` key is missing from the merged AnnData object. However, when I replace the `None` value with a value like `-1`, it works fine. Is this the expected behavior, and if so why?

**Comment by ivirshup** [comment]
Thanks for the bug report, and sorry about the time it took to respond.

I'm not able to reproduce with either 0.7.8 or 0.8.0. Are you still seeing this?

```python
import anndata as ad
import numpy as np

a = ad.AnnData(np.ones((3, 3)), uns={"property": {"0": None}})
b = ad.AnnData(np.ones((3, 3)), uns={"property": {"1": None}})
ad.concat([a, b], uns_merge="unique").uns["property"]
# {'0': None, '1': None}
```

```python
ad.__version__
# '0.7.8'
```

**Comment by alam-shahul** [comment]
Okay, I guess I reported the bug slightly incorrectly. The result is correct after concatenating, but after I write to an HDF5 file with `merged_datast.write(filepath)` and then reload with `reloaded_dataset = ad.read_h5ad(filepath)`, it does not contain the right keys. Is this expected?

```
import anndata as ad
import numpy as np
a = ad.AnnData(np.ones((3, 3)), uns={"property": {"0": None}})
b = ad.AnnData(np.ones((3, 3)), uns={"property": {"1": None}})


merged_dataset = ad.concat([a, b], uns_merge="unique")
merged_dataset.uns["property"]
# {'0': None, '1': None}

merged_dataset.write("issue.h5")

reloaded_dataset = ad.read_h5ad("issue.h5")
reloaded_dataset.uns["property"]
# {}

ad.__version__
# '0.8.0'
```

**Comment by flying-sheep** [comment]
Will be fixed in #999
